src/ir/fake_predicate_arena.h

```diff
@@ -0,0 +1,38 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#ifndef SRC_IR_FAKE_PREDICATE_ARENA_H_
+#define SRC_IR_FAKE_PREDICATE_ARENA_H_
+
+#include "src/ir/predicate_arena.h"
+
+namespace raksha::ir {
+
+// An "arena" that just returns the objects it should capture, releasing them
+// from the std::unique_ptr, which leaks them. Allows avoiding mutable global
```

Comment:
Even for tests, leaks seem like a bad idea. How about we make the arena part of the `Test` context and use it?

Reply:
You mean like a member of a test fixture? The issue here is that I'm using this arena to create global test objects to place into a list of parameters for a `TEST_P` pattern. As such, they need to be constructed before the test fixture.

In a previous draft, I tried making these test objects be lambdas that would yield the test object when provided the arena. This allowed me to parameterize the `TEST_P` with the lambdas and place the arena inside the test fixture. That was nice from a statelessness perspective, but it was a pretty deep change to the organization of the tests. I can switch back to that approach.

Reply:
Yes, I meant in a text fixture. Getting rid of the fake arena would be nice, but not the end of the world. I will let you make the call on whether it is worth the change in the organization of the tests.

Reply:
I've come up with a lower-cost way of doing it: make a test-only class that contains a predicate and the single-use arena used to allocate it. This will allow easy construction of const test objects in the static lifetime while also avoiding mutable state. Going to do that.

Reply:
Fixed.